summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SpoutProvider.scala

```diff
@@ -147,12 +143,15 @@ case class SpoutProvider(storm: Storm, stormDag: Dag[Storm], node: StormNode, jo
    * This is the decision logic.
    * If the spout is followed by a summer wraps the spout in KeyValueSpout
    */
-  def apply: IRichSpout = {
-    val tormentaSpout = computerSpout()
+  def apply: (Int, IRichSpout) = {
+    val (spout, sourceParallelismOption) = extractSourceAttributes
+    val tormentaSpout = computerSpout(spout)
+    val sourceParallelism = getSourceParallelism(sourceParallelismOption)
     val summerOpt: Option[SummerNode[Storm]] = stormDag.dependantsOf(node.asInstanceOf[StormNode]).collect { case s: SummerNode[Storm] => s }.headOption
-    summerOpt match {
+    val stormSpout: IRichSpout = summerOpt match {
       case Some(summerNode) => getKeyValueSpout[Any, Any](tormentaSpout.asInstanceOf[Spout[(Timestamp, (Any, Any))]], summerNode)
```

Comment:
can we move the cast here:
```scala
case Some(summerNode) =>
  // since we know that the the next node is a summer, this implies the type of the spout must be `(K, V)` for some `K, V`
  val kvSpout = tormentaSpout.asInstanceOf[Spout[(Timestamp, (Any, Any))]]
  getKeyValueSpout(kvSpout, summerNode)
```